include/avk/semaphore.hpp

```diff
@@ -60,6 +75,377 @@ namespace avk
 		std::vector<any_owning_resource_t> mLifetimeHandledResources;
 	};
 
+	/**
+	 * @brief This is an explicit template specialization of owning_resource as defined in cpp_utils.hpp
+	 * It is a complete duplicate of the class that contains an additional overload for operator= (defined at the bottom).
+	 *
+	 * The definition of an explicit template specialization is required since operator= overloads cannot be defined as non-member functions.
+	 * (see https://en.cppreference.com/w/cpp/language/operators)
+	 * As far as I can tell, defining an explicit template specialization for a class requires duplicating the entire class even if only minor changes are made.
+	 * 
+	 *
+	 * The overload allows instantiating semaphore_value_info structs with the expression `sem = val`.
+	 * Where the type of sem is `owning_resource<semaphore_t>` and the type of val is `uint64_t`.
+	 */
+	template<>
+	class owning_resource<semaphore_t> : public std::variant<std::monostate, semaphore_t, std::shared_ptr<semaphore_t>>

```

Comment:
Here I am defining an explicit template specialization for `owning_resource<semaphore_t>` (and `resource_argument<semaphore_t>` below) in order to add an overload for the `=` operator. I can't simply use non-member operator overloads (like for `>=`) since `operator=` explicitly [cannot be declared as non-member function](https://en.cppreference.com/w/cpp/language/operators).

This results in a lot of duplicated code, since the remaining functions need to be redefined as well. As far as I am aware it isn't possible to define new member functions ( / operator overloads) for a specific template configuration without redefining the entire class, though I still hope I've missed something in the [documentation](https://en.cppreference.com/w/cpp/language/template_specialization).